
 <group ns="rtabmap">




    <!-- Use RGBD synchronization -->
    <!-- Here is a general example using a standalone nodelet, 
         but it is recommended to attach this nodelet to nodelet 
         manager of the camera to avoid topic serialization -->
    <node pkg="rtabmap_ros" type="rgbd_sync" name="rgbd_sync" output="screen">
      <!--remap from="/thorvald_001/kinect2_front_camera/rgb/image_raw" to="/camera/rgb/image_raw"/>
      <remap from="/thorvald_001/kinect2_front_sensor/ir/depth/image_raw" to="/camera/depth/image_raw"/>
      <remap from="/thorvald_001/kinect2_front_camera/rgb/camera_info" to="/camera/rgb/camera_info"/>
      <remap from="/thorvald_001/odometry/base_raw" to="/odom"/>
      <remap from="/thorvald_001/front_scan"        to="/scan"/-->

      <remap from="rgbd_image"      to="rgbd_image"/> <!-- output -->
      
      <!-- Should be true for not synchronized camera topics 
           (e.g., false for kinectv2, zed, realsense, true for xtion, kinect360)-->
      <param name="approx_sync"       value="false"/> 
    </node>

    <node name="rtabmap" pkg="rtabmap_ros" type="rtabmap" output="screen" args="--delete_db_on_start">
          <param name="database_path"       type="string" value="~/.ros/rtabmap.db"/>
          <param name="frame_id" type="string" value="thorvald_001/base_link"/>
          <param name="subscribe_rgb"   type="bool" value="true"/>
          <param name="subscribe_depth" type="bool" value="true"/>
          <param name="subscribe_rgbd" type="bool" value="false"/>
          <param name="subscribe_scan" type="bool" value="true"/>


          <remap from="rgbd_image" to="rgbd_image"/>

          <param name="queue_size" type="int" value="100"/>

				  <!-- output -->
				  <remap from="grid_map" to="/map"/>
			
          <!-- RTAB-Map's parameters -->
          <param name="RGBD/NeighborLinkRefining" type="string" value="true"/>
          <param name="RGBD/ProximityBySpace"     type="string" value="true"/>
          <param name="RGBD/AngularUpdate"        type="string" value="0.01"/>
          <param name="RGBD/LinearUpdate"         type="string" value="0.01"/>
          <param name="RGBD/OptimizeFromGraphEnd" type="string" value="false"/>
          <param name="Grid/FromDepth"            type="string" value="false"/> <!-- occupancy grid from lidar -->
          <param name="Reg/Force3DoF"             type="string" value="true"/>
          <param name="Reg/Strategy"              type="string" value="1"/> <!-- 1=ICP -->
          
          <!-- ICP parameters -->
          <param name="Icp/VoxelSize"                 type="string" value="0.05"/>
          <param name="Icp/MaxCorrespondenceDistance" type="string" value="0.1"/>
    </node>

    <!-- visualization with rtabmapviz -->
    <node pkg="rtabmap_ros" type="rtabmapviz" name="rtabmapviz" args="-d $(find rtabmap_ros)/launch/config/rgbd_gui.ini" output="screen">
      <param name="subscribe_scan"   type="bool"   value="true"/>
      <param name="subscribe_odom"   type="bool"   value="true"/>
      <param name="frame_id"         type="string" value="thorvald_001/base_link"/>
      <param name="approx_sync"      type="bool"   value="true"/>

      <remap from="odom"            to="/odom"/>
      <remap from="scan"            to="/scan"/>
    </node>

  </group>
